### PR TITLE
GH-1634 javadoc aggregate

### PR DESCRIPTION
**Pull request by jeenbroekstra**
GitHub issue resolved: #1634   

Briefly describe the changes proposed in this PR:

* javadoc aggregate creation has been moved to root pom as per https://maven.apache.org/plugins/maven-javadoc-plugin/examples/aggregate.html
* removed obsolete assembly profile from core module pom
* release script modified to handle simpler way to generate javadoc and SDK zip.

### CODE REVIEW

File: release.sh [hunk 1]
```diff
@@ -258,5 +255,6 @@ echo " - Edit the following file https://github.com/eclipse/rdf4j-doc/blob/maste
 echo " - Go to https://github.com/eclipse/rdf4j-doc/tree/master/site/content/news and create rdf4j-${NEWS_FILE_NAME}.md"
 echo " - Post to Google Groups: https://groups.google.com/forum/#!forum/rdf4j-users"
 echo "     - Good example: https://groups.google.com/forum/#!topic/rdf4j-users/isrC7qdhplY"
-echo " - Upload the javadocs " #TODO Where should they be uploaded and how?
+echo " - Upload the javadocs by adding them to rdf4j-doc project: site/static/javadoc/${MVN_VERSION_RELEASE}"
```
Comment (hmottestad) [inline release.sh]: The location of the generated javadocs, is it: assembly/target/site/apidocs

Comment (jeenbroekstra) [reply release.sh]: Actually it's target/site/apidocs (in the project root) now. Or, of course, you can get them by unzipping the SDK zip :)

Comment (hmottestad) [reply release.sh]: Could you add that as an echo line to the script? In the future we could automate that too.

**Comment by hmottestad** [comment]
Lgtm

**Pull request merged**